src/playerhud.py

```diff
@@ -48,6 +53,11 @@ def __init__(
         self.health_sprite.center_x = hud_x
         self.health_sprite.center_y = hud_y
 
+        if isinstance(self.player.game_mode_state, StockGameModePlayerState):
```

Comment:
this kind of `if isinstance(...)` logic looks like a really good candidate for refactoring into subclasses or composed methods